modules/zones/main.tf

```
@@ -1,23 +1,23 @@
 resource "aws_route53_zone" "this" {
   for_each = { for k, v in var.zones : k => v if var.create }
 
-  name          = lookup(each.value, "domain_name", each.key)
-  comment       = lookup(each.value, "comment", null)
-  force_destroy = lookup(each.value, "force_destroy", false)
+  name          = coalesce(each.value.domain_name, each.key)
+  comment       = each.value.comment
+  force_destroy = each.value.force_destroy
 
-  delegation_set_id = lookup(each.value, "delegation_set_id", null)
+  delegation_set_id = each.value.delegation_set_id
 
   dynamic "vpc" {
-    for_each = try(tolist(lookup(each.value, "vpc", [])), [lookup(each.value, "vpc", {})])
+    for_each = each.value.vpc
```

not quite, I think we need more guards/checks

```suggestion
    for_each = each.value.vpc != null ? each.value.vpc : []
```